### ISSUE TITLE
Export Files playlist dialog is broken

### ISSUE THREAD

**Issue by javiercantero** [exaile/exaile#299]
The dialog to export files from a playlist (launched with the 'Export Files' context menu option over such playlist tab) fails with this error:
```
Traceback (most recent call last):
  File ".../exaile/xlgui/widgets/dialogs.py", line 1614, in <lambda>
    dialog.connect( 'uris-selected', lambda widget, uris: _on_uri(uris[0]))
  File ".../exaile/xlgui/widgets/dialogs.py", line 1604, in _on_uri
    if hasattr(playlist, 'get_playlist'):
UnboundLocalError: local variable 'playlist' referenced before assignment
```
This is happening since the commit 1a8eb62 that allowed exporting of SmartPlaylists  (#216).

**Comment by javiercantero** [comment]
Ok, I found it: in [line 1605](https://github.com/exaile/exaile/blob/master/xlgui/widgets/dialogs.py#L1605) of `xlgui/widgets/dialogs.py` the commit tries to modify the free variable `playlist` of the `_on_uri()` closure, but in Python the binding of a free variable in a closure is **read-only**. The solution is simple: don't assign `playlist`.

**Comment by javiercantero** [comment]
Fixed with #300
